runtests.py

```diff
@@ -325,14 +327,18 @@ def build_project(args):
 
     # Always use ccache, if installed
     env['PATH'] = os.pathsep.join(EXTRA_PATH + env.get('PATH', '').split(os.pathsep))
-
+    cvars = distutils.sysconfig.get_config_vars()
+    if 'gcc' in cvars['CC']:
+        # add flags used as werrors tools/travis-test.sh
+        warnings_as_errors = (' -Werror=declaration-after-statement -Werror=vla '
+                             '-Werror=nonnull -Werror=pointer-arith '
```

Comment:
fixed

Reply:
IIRC, the `-Wno-sign-compare` was added so that things would work on travis, as the check was added in Python 3.6 (?). We may want to see if it is still needed, I fixed a lot of those cases.

Reply:
This may be a problem for `-Wno-signed-compare` if it overrides the local environment. What if one wants to enable the flag to check for problems?

Reply:
I will reverse the order, so these flags will be overridden by the local environment, and will remove that flag from here. TBD if we can remove it from the `travis-test.sh` script on all python versions

Reply:
I suppose we could also add a `'cflags'` command line variable.

Reply:
instead of a command line flag, I just do `CFLAGS='-O0 -g' python runtests.py`. the PR does support that syntax.